api/helpers.go

```diff
@@ -111,13 +111,38 @@ func getRedirectTo(r *http.Request) (reqref string) {
 	return
 }
 
+func parseURL(u string) (*url.URL, error) {
+	parsed, err := url.Parse(u)
+	if err != nil {
+		return nil, err
+	}
+
+	if parsed.Scheme == "https" || parsed.Scheme == "http" {
+		// this normalizes URLs that end with just a `/` as you would
+		// get from window.location.href on the frontend, so that they
+		// would match glob rules correctly
+
+		// applied only http and https URLs as trailing slashes don't
+		// change the behavior, while in other schemes they may have
+		// another meaning
+
+		// see: https://en.wikipedia.org/wiki/URI_normalization#Normalizations_that_usually_preserve_semantics
+
+		parsed.Path = strings.TrimSuffix(parsed.Path, "/")
+	}
+
+	return parsed, nil
+}
+
 func isRedirectURLValid(config *conf.Configuration, redirectURL string) bool {
 	if redirectURL == "" {
 		return false
 	}
 
-	base, berr := url.Parse(config.SiteURL)
-	refurl, rerr := url.Parse(redirectURL)
+	base, berr := parseURL(config.SiteURL)
```

Comment:
we can move this to when we load the config for `SiteURL` too rather than having to do it each time we check if the redirect url is valid

Reply:
Hm, yeah probably best there... but would have to make some of these methods public I guess.

Reply:
you can consider adding it in [ApplyDefaults](https://github.com/supabase/gotrue/blob/master/conf/configuration.go#L279) instead of having it as a separate function